lib/fog/aws/storage.rb

```diff
@@ -278,15 +278,13 @@ def request_params(params)
             if params[:bucket_cname]
               host = bucket_name
             else
-              path_style = params.fetch(:path_style, @path_style)
-              if !path_style
-                if COMPLIANT_BUCKET_NAMES !~ bucket_name
-                  Fog::Logger.warning("fog: the specified s3 bucket name(#{bucket_name}) is not a valid dns name, which will negatively impact performance.  For details see: http://docs.amazonwebservices.com/AmazonS3/latest/dev/BucketRestrictions.html")
-                  path_style = true
-                elsif scheme == 'https' && !path_style && bucket_name =~ /\./
-                  Fog::Logger.warning("fog: the specified s3 bucket name(#{bucket_name}) contains a '.' so is not accessible over https as a virtual hosted bucket, which will negatively impact performance.  For details see: http://docs.amazonwebservices.com/AmazonS3/latest/dev/BucketRestrictions.html")
-                  path_style = true
-                end
+              path_style = false
+              if COMPLIANT_BUCKET_NAMES !~ bucket_name
+                Fog::Logger.warning("fog: the specified s3 bucket name(#{bucket_name}) is not a valid dns name, which will negatively impact performance. For details see: http://docs.amazonwebservices.com/AmazonS3/latest/dev/BucketRestrictions.html")
```

Comment:
This looks like a good approach based on our discussion. Maybe we should tweak the warning messages to refer to deprecation and link to the resources we found though? Or maybe even just link to the issue where we discussed this so people can get context (especially since it has the other links). What do you think?

Reply:
I was thinking about this too, the existing link does give details on requirements for bucket names. But I think linking to the github issue is probably the best option.

Reply:
Sounds good to me. We can certainly add a reference in the issue to this bucket restrictions page for reference also.